File: Ruby/CriterioAceptacion.rb
require('./HistoriaUsuario.rb');

public
class CriterioAceptacion

	# @ReturnType int
	public
	def getId()
		return @_id
	end

	# @ParamType aId int
	# @ReturnType void
	public
	def setId(aId)
		@_id = aId
	end

	# @ReturnType int
	public
	def getPrioridad()
		return @_prioridad
	end

	# @ParamType aPrioridad int
	# @ReturnType void
	public
	def setPrioridad(aPrioridad)
		@_prioridad = aPrioridad
	end

	# @ReturnType String
	public
	def getContexto()
		return @_contexto
	end

	# @ParamType aContexto String
	# @ReturnType void
	public
	def setContexto(aContexto)
		@_contexto = aContexto
	end

	# @ReturnType String
	public
	def getEvento()
		return @_evento
	end

	# @ParamType aEvento String
	# @ReturnType void
	public
	def setEvento(aEvento)
		@_evento = aEvento
	end

	# @ReturnType String
	public
	def getResultado()
		return @_resultado
	end

	# @ParamType aResultado String
	# @ReturnType void
	public
	def setResultado(aResultado)
		@_resultado = aResultado
	end

	public
	def criterioAceptacion()
		# Not yet implemented
	end

	public
	def initialize()
		# @AttributeType int
		@_id
		# @AttributeType int
		@_prioridad
		# @AttributeType String
		@_contexto
		# @AttributeType String
		@_evento
		# @AttributeType String
		@_resultado
		# @AttributeType HistoriaUsuario
		# # @AssociationType HistoriaUsuario
		# # @AssociationMultiplicity 1..*
		@_tiene = Array.new()

		# Not yet implemented
	end
end

File: C#/CriterioAceptacion.cs
using System;

public class CriterioAceptacion {
	private int id;
	private int prioridad;
	private String contexto;
	private String evento;
	private String resultado;

	public int GetId() {
		return this.id;
	}
	public void SetId(ref int id) {
		this.id = id;
	}
	public int GetPrioridad() {
		return this.prioridad;
	}
	public void SetPrioridad(ref int prioridad) {
		this.prioridad = prioridad;
	}
	public String GetContexto() {
		return this.contexto;
	}
	public void SetContexto(ref String contexto) {
		this.contexto = contexto;
	}
	public String GetEvento() {
		return this.evento;
	}
	public void SetEvento(ref String evento) {
		this.evento = evento;
	}
	public String GetResultado() {
		return this.resultado;
	}
	public void SetResultado(ref String resultado) {
		this.resultado = resultado;
	}
	public CriterioAceptacion() {
		throw new System.NotImplementedException("Not implemented");
	}

	private HistoriaUsuario[] tiene;

}
